File: sdks/python/examples/smoke_test.py
#!/usr/bin/env python3
"""Smoke example covering the learning-path flow."""

from __future__ import annotations

import sys
import time
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1] / "src"))

from notification_hub import NotificationClient, NotificationHubError


def main() -> int:
    api_key = sys.argv[1] if len(sys.argv) > 1 else "secure-token-123"
    base_url = sys.argv[2] if len(sys.argv) > 2 else "http://localhost:3000"

    print("Initializing Notification SDK client...")
    client = NotificationClient(api_key, base_url)
    template_id = f"sdk-demo-{int(time.time())}"

    try:
        print("\n--- Create template ---")
        created = client.create_template(template_id, "SDK Demo", "email", body="Hello {{name}}")
        print(f"Created: {created.id}")

        print("\n--- Retrieve template ---")
        retrieved = client.get_template(created.id)
        print(f"Retrieved: {retrieved.name}")

        print("\n--- Set preference ---")
        prefs = client.set_preferences("sdk@example.com", email=True, sms=False, push=False)
        print(f"Preferences: {prefs}")

        print("\n--- Send notification ---")
        response = client.send_notification(
            "sdk@example.com",
            "email",
            created.id,
            {"name": "Alexander"},
        )
        print(f"Sent: {response}")

        print("\n--- Check record status ---")
        record = client.get_record(response.recordId)
        print(f"Status: {record.status}")

        print("\nAll SDK integration tests completed successfully!")
        return 0
    except NotificationHubError as exc:
        print("\nAn error occurred during SDK execution test:", file=sys.stderr)
        print(exc, file=sys.stderr)
        return 1


if __name__ == "__main__":
    raise SystemExit(main())

File: sdks/java/src/test/java/com/sophias/hub/notification/SmokeTest.java
package com.sophias.hub.notification;

import java.util.Map;

/**
 * Smoke example covering the learning-path flow.
 *
 * <pre>
 * mvn -q test-compile exec:java
 * </pre>
 */
public final class SmokeTest {
  private SmokeTest() {}

  public static void main(String[] args) {
    String apiKey = args.length > 0 ? args[0] : "secure-token-123";
    String baseUrl = args.length > 1 ? args[1] : "http://localhost:3000";

    System.out.println("Initializing Notification SDK client...");
    NotificationClient client = new NotificationClient(apiKey, baseUrl);
    String templateId = "sdk-demo-" + System.currentTimeMillis();

    try {
      System.out.println("\n--- Create template ---");
      NotificationTemplate template = new NotificationTemplate();
      template.setId(templateId);
      template.setName("SDK Demo");
      template.setChannel("email");
      template.setBody("Hello {{name}}");
      NotificationTemplate created = client.createTemplate(template);
      System.out.println("Created: " + created.getId());

      System.out.println("\n--- Retrieve template ---");
      NotificationTemplate retrieved = client.getTemplate(created.getId());
      System.out.println("Retrieved: " + retrieved.getName());

      System.out.println("\n--- Set preference ---");
      Preferences prefs = client.setPreferences("sdk@example.com", true, false, false);
      System.out.println("Preferences email=" + prefs.isEmail());

      System.out.println("\n--- Send notification ---");
      NotificationResponse response = client.sendNotification(
          "sdk@example.com",
          "email",
          created.getId(),
          Map.of("name", "Alexander")
      );
      System.out.println("Sent: " + response);

      System.out.println("\n--- Check record status ---");
      DeliveryRecord record = client.getRecord(response.getRecordId());
      System.out.println("Status: " + record.getStatus());

      System.out.println("\nAll SDK integration tests completed successfully!");
    } catch (NotificationHubException e) {
      System.err.println("\nAn error occurred during SDK execution test:");
      System.err.println(e.getMessage());
      System.exit(1);
    }
  }
}
